File: typescript/src/sliding-window/maximum-average-subarray.ts
export class MaximumAverageSubarray {
  findMaxAverage(nums: number[], k: number): number {
    let windowSum = nums.slice(0, k).reduce((sum, value) => sum + value, 0);
    let bestSum = windowSum;

    for (let right = k; right < nums.length; right++) {
      windowSum += nums[right] - nums[right - k];
      bestSum = Math.max(bestSum, windowSum);
    }

    return bestSum / k;
  }
}

File: python/solutions/sliding_window/maximum_average_subarray.py
from typing import List

class Solution:
    def findMaxAverage(self, nums: List[int], k: int):
        window_sum = sum(nums[:k])
        best_sum = window_sum

        for right in range(k, len(nums)):
            window_sum += nums[right] - nums[right - k]
            best_sum = max(best_sum, window_sum)

        return best_sum / k
